msi/GitProduct.wxs

```diff
@@ -27,13 +28,64 @@
+      <RegistrySearch Id="TerminalSearch" Root="HKLM" Key="SOFTWARE\GitForWindows" Name="Terminal" Type="raw" />
+    </Property>
+    <SetProperty Id="CMDLINE_TERMINAL" Value="[TERMINAL]" Before="AppSearch" Sequence="first" />
+    <SetProperty Id="TERMINAL" Value="[CMDLINE_TERMINAL]" After="AppSearch" Sequence="first">CMDLINE_TERMINAL</SetProperty>
+
+    <!-- ENABLEFSCACHE Property implementing the "Remember Property" pattern -->
+    <Property Id="ENABLEFSCACHE" Value="1">
+      <RegistrySearch Id="EnableFsCacheSearch" Root="HKLM" Key="SOFTWARE\GitForWindows" Name="EnableFsCache" Type="raw" />
+    </Property>
+    <SetProperty Id="CMDLINE_ENABLEFSCACHE" Value="[ENABLEFSCACHE]" Before="AppSearch" Sequence="first" />
+    <SetProperty Id="ENABLEFSCACHE" Value="[CMDLINE_ENABLEFSCACHE]" After="AppSearch" Sequence="first">CMDLINE_ENABLEFSCACHE</SetProperty>
+
+    <!-- ENABLEGCM Property implementing the "Remember Property" pattern -->
+    <Property Id="ENABLEGCM" Value="1">
+      <RegistrySearch Id="EnableGcmSearch" Root="HKLM" Key="SOFTWARE\GitForWindows" Name="EnableGcm" Type="raw" />
+    </Property>
+    <SetProperty Id="CMDLINE_ENABLEGCM" Value="[ENABLEGCM]" Before="AppSearch" Sequence="first" />
+    <SetProperty Id="ENABLEGCM" Value="[CMDLINE_ENABLEGCM]" After="AppSearch" Sequence="first">CMDLINE_ENABLEGCM</SetProperty>
+
     <Feature Id='Git'>
       <Component Directory='INSTALLFOLDER'>
-        <RegistryValue Root='HKLM' Key='Software\Git' Name='InstallFolder' Value='[INSTALLFOLDER]' Type='string' />
+        <RegistryValue Root='HKLM' Key='SOFTWARE\GitForWindows' Name='InstallFolder' Value='[INSTALLFOLDER]' Type='string' />
         <Environment Name='PATH' Value='[BinFolder]' Action='set' Part='last' System='yes' />
-
         <RemoveFolder Directory='GitProgramMenuFolder' On='uninstall' />
       </Component>
+      <Component Directory='ApplicationFolder'>
+        <RegistryValue Action='write' Type='string' Root='HKLM' Key='SOFTWARE\GitForWindows' Name='CliEnvironment' Value='[CLIENVIRONMENT]' />
+        <RemoveFolder Id='RemoveApplicationFolder' On='uninstall' />
+      </Component>
+      <Component Directory='ApplicationFolder'>
+        <RegistryValue Action='write' Type='string' Root='HKLM' Key='SOFTWARE\GitForWindows' Name='LineEndings' Value='[LINEENDINGS]' />
+      </Component>
+      <Component Directory='ApplicationFolder'>
+        <RegistryValue Action='write' Type='string' Root='HKLM' Key='SOFTWARE\GitForWindows' Name='Terminal' Value='[TERMINAL]' />
+      </Component>
+      <Component Directory='ApplicationFolder'>
+        <RegistryValue Action='write' Type='integer' Root='HKLM' Key='SOFTWARE\GitForWindows' Name='EnableFsCache' Value='[ENABLEFSCACHE]' />
+      </Component>
+      <Component Directory='ApplicationFolder'>
+        <RegistryValue Action='write' Type='integer' Root='HKLM' Key='SOFTWARE\GitForWindows' Name='EnableGcm' Value='[ENABLEGCM]' />
+      </Component>
```

Comment:
As can be seen at [this link](https://msdn.microsoft.com/en-us/library/windows/desktop/aa372105(v=vs.85).aspx), each installed program has a unique product code, which MUST be represented by a GUID. There are certain [registry entries](https://msdn.microsoft.com/en-us/library/windows/desktop/aa372105(v=vs.85).aspx) that are created in the product's uninstall registry key which allow for information to be displayed in the Add/Remove Programs a/k/a Programs and Features Windows Control Panel applet (from here on out, ARP).

When following the product code guidelines, then, it's possible that two different versions of the same product may have different product codes. Therefore, when following the product code guidelines, one cannot be sure that a given product's uninstall information resides in the same registry location from version to version.

The Git for Windows InnoSetup installer does not currently specify an `AppId`. Therefore, InnoSetup uses the `AppName` to generate the uninstall key name -- in this case, `Git_is` (i.e. `{AppName}_is`). This behavior does not conform to the product code guidelines above and could potentially cause problems (though thus far, it doesn't seem to have created any apparent ones that anyone is aware of).

In WiX, it's not possible to specify an uninstall key name that is not the product code GUID (represented by the `@Id` attribute of the `<Product />` element) because WiX follows the product code guidelines as much as it can. (In the end, the installer author is still ultimately responsible for ensuring that the product code is changed per the guidelines when it is required).

Because of the product code guidelines, then, metadata about installed programs should be located under `HKLM\SOFTWARE\<ProductName>` or `HKCU\SOFTWARE\<ProductName>` (depending on installation context) and 3rd party applications that depend on knowing information about another installed product should look under those locations for that information.

Now, the current InnoSetup installer does _also_ create a `CurrentVersion` registry value in the `HKLM\SOFTWARE\GitForWindows` key, and so likewise, I'm creating that registry value entry in the same location via the MSI installer. 3rd party applications that depend on metadata about the current installation of Git for Windows should look in this location for that information.